import/index_java/src/org/vufind/index/ConfigManager.java

```diff
@@ -210,55 +216,42 @@ public Ini loadConfigFile(String filename)
                 retVal.put(key, overrideSection.get(key));
             }
         }
+
         return retVal;
     }
 
     /**
-     * Get a setting from a VuFind configuration file.
+     * Get a setting from a VuFind configuration file and sanitize the value.
      * @param filename configuration file name
      * @param section section name within the file
      * @param setting setting name within the section
      */
-    public String getConfigSetting(String filename, String section, String setting)
+    public String getSanitizedConfigSetting(String filename, String section, String setting)
     {
-        String retVal = null;
-
-        // Grab the ini file.
-        Ini ini = loadConfigFile(filename);
-
-        // Check to see if we need to worry about an override file:
-        String override = ini.get("Extra_Config", "local_overrides");
-        if (override != null) {
-            Ini overrideIni = loadConfigFile(override);
-            retVal = overrideIni.get(section, setting);
-            if (retVal != null) {
-                return sanitizeConfigSetting(retVal);
-            }
+        String retVal = getConfigSetting(filename, section, setting);
+        if (retVal == null) {
+            logger.warn(section + "." + setting + " setting missing from " + filename);
+            return retVal;
         }
 
-        // Try to find the requested setting:
-        retVal = ini.get(section, setting);
-
-        //  No setting?  Check for a parent configuration:
-        while (retVal == null) {
-            String parent = ini.get("Parent_Config", "path");
-            if (parent !=  null) {
-                try {
-                    ini.load(new FileReader(new File(parent)));
-                } catch (Throwable e) {
-                    dieWithError(
-                        "Unable to access " + parent
-                        + " (" + e.getMessage() + ")"
-                    );
-                }
-                retVal = ini.get(section, setting);
-            } else {
-                break;
-            }
-        }
+        // Return a copy of the string.
+        // We do not want the sanitizer to update the cache, because it might
+        // cause problems when executing them multiple times, like
+        // e.g. in multithreaded scenarios.
+        String retValCopy = new String(retVal);
+        return sanitizeConfigSetting(retValCopy);
```

Comment:
Would it be better to move the string copying logic inside sanitizeConfigSetting so we don't need to do it in two places? Since sanitizeConfigSetting returns a value, I don't think it was actually my intention for it to do in-place modification of values -- I just didn't realize that the argument was being treated as a reference.

Reply:
Yeah, I agree that copying strings in sanitizeConfigSetting is not enough by itself to solve the problem; you still need to copy the section in getSanitizedConfigSection(). However, if you copy the string in sanitizeConfigSetting, then you don't have to copy the values in getSanitizedConfigSetting and getSanitizedConfigSection, and you could in fact probably reduce the end of getSanitizedConfigSetting to simply `return retVal == null ? retVal : sanitizeConfigSetting(retVal);`

I don't have really strong feelings about this if you don't want to change it further, but it still feels like it might be a worthwhile simplification.

Reply:
Done!